src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java

```diff
@@ -100,11 +101,26 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
         final PrintStream logger = taskListener.getLogger();
 
         try {
-            final AppCenterServiceFactory appCenterServiceFactory = new AppCenterServiceFactory(
-                    getApiToken(), getOwnerName(), getAppName(), getPathToApp(), getBaseUrl()
-            );
-
-            return filePath.act(new UploadTask(filePath, taskListener, appCenterServiceFactory));
+        	EnvVars vars = run.getEnvironment(taskListener);
+        	String pathParsed = vars.expand(getPathToApp());
+        	logger.println("Path is " + pathParsed);
+        	FilePath localFiles[] = filePath.list(pathParsed);
+        	if(localFiles.length == 0) {
+        		logger.println("No file found to upload: " + pathParsed);
```

Comment:
1. We do this check in `UploadTask`.
2. We want to move as much business logic as possible away from Jenkins and into the tasks themselves (easier to test).